traffic_ops/testing/api/v14/serverservercapability_test.go

```diff
@@ -0,0 +1,165 @@
+package v14
+
+/*
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"testing"
+
+	"github.com/apache/trafficcontrol/lib/go-log"
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-util"
+)
+
+func TestServerServerCapabilities(t *testing.T) {
+	WithObjs(t, []TCObj{CDNs, Types, Parameters, Profiles, Statuses, Divisions, Regions, PhysLocations, CacheGroups, Servers, ServerServerCapabilities}, func() {
+		GetTestServerServerCapabilities(t)
+	})
+}
+
+func CreateTestServerServerCapabilities(t *testing.T) {
+
+	// Valid POSTs
+
+	// loop through server ServerCapabilities, assign FKs and create
+	for _, ssc := range testData.ServerServerCapabilities {
+		servResp, _, err := TOSession.GetServerByHostName(*ssc.Server)
+		if err != nil {
+			t.Fatalf("cannot GET Server by hostname: %v - %v\n", *ssc.Server, err)
+		}
+		if servResp == nil || len(servResp) != 1 {
+			t.Fatalf("cannot GET Server by hostname: %v. Response did not include record.\n", *ssc.Server)
+		}
+		server := servResp[0]
+		ssc.ServerID = &server.ID
+		resp, _, err := TOSession.CreateServerServerCapability(ssc)
+		if err != nil {
+			t.Errorf("could not POST the server capability %v to server %v: %v\n", *ssc.ServerCapability, *ssc.Server, err)
+		}
+		log.Debugln("Response: ", server.HostName, " ", resp)
+	}
+
+	// Invalid POSTs
+
+	ssc := testData.ServerServerCapabilities[0]
+
+	// Attempt to assign already assigned server capability
+	_, _, err := TOSession.CreateServerServerCapability(ssc)
+	if err == nil {
+		t.Errorf("expected to receive error when assigning a already assigned server capability\n")
```

Comment:
Consider using `t.Error` since there's no interpolation.

I believe the trailing `\n` isn't needed since the test framework will print each error on its own line by default. Although adding it seems to be a project convention.

Reply:
Some things just propagate through copy/paste of existing code (I expect the `\n` thing is has just propagated due to not knowing it's unnecessary -- a lot of us were inexperienced with Go when the whole Go effort in trafficcontrol began).